### ISSUE TITLE
[BUG] VM not rescheduled automatically when node goes down

### ISSUE THREAD

**Issue by janeczku** [harvester/harvester#1255]
Tracking the known issue where VMs are not rescheduled automatically in case a node goes down. We also should document the expected behaviour.

**Comment by w13915984028** [comment]
a quick verification in v1.0.0-rc2:
(1) manually shutdown Harvester NODE (based on KVM VM), which is one of the master node, has one VM running on it.

![image](https://user-images.githubusercontent.com/31133476/148234653-d9d527df-bec8-469c-a98c-9abd92a039b6.png)

(2) the NODE became "unavailable" in Harvester, Hosts
![image](https://user-images.githubusercontent.com/31133476/148234545-aaa54493-ecbc-42dd-a813-7afd5345edd2.png)

(3) the vm, which runs in the "Unavailable" Node at beginning, is rescheduled to other Node
![image](https://user-images.githubusercontent.com/31133476/148235669-ca2975a9-6812-4501-9e75-a02ad4c9df99.png)

(4) in my ENV, following errors are encountered, which cause the VM restarting failed
![image](https://user-images.githubusercontent.com/31133476/148235607-ca23b512-5351-4027-929f-dd940357b7a0.png)

`AttachVolume.Attach failed for volume "pvc-949549be-92b2-4105-b1ca-5e7d94ee6967" : rpc error: code = Internal desc = Bad response statusCode [500]. Status [500 Internal Server Error]. Body: [code=Server Error, detail=, message=unable to attach volume pvc-949549be-92b2-4105-b1ca-5e7d94ee6967 to harv2: volume must be healthy or degraded to start migration] from [http://longhorn-backend:9500/v1/volumes/pvc-949549be-92b2-4105-b1ca-5e7d94ee6967?action=attach]`

`AttachVolume.Attach failed for volume "pvc-c796c818-6b26-4985-bb8a-e4eecdeca80b" : rpc error: code = Internal desc = Bad response statusCode [500]. Status [500 Internal Server Error]. Body: [code=Server Error, detail=, message=unable to attach volume pvc-c796c818-6b26-4985-bb8a-e4eecdeca80b to harv2: volume must be healthy or degraded to start migration] from [http://longhorn-backend:9500/v1/volumes/pvc-c796c818-6b26-4985-bb8a-e4eecdeca80b?action=attach]
`

**Comment by w13915984028** [comment]
@janeczku 
Could you please confirm:
From your issue description, does the auto rescheduling of VM in harvester v1.0.0 meet your expectation? When feasible, please also check the latest version in your ENV. 

When issue still exists, please write down the operation steps, thanks.

**Comment by w13915984028** [comment]
update: 
the VM finally runs successfully in NODE harv2.   (it reported failure in previous update)

![image](https://user-images.githubusercontent.com/31133476/148247867-29082509-9962-40d6-8036-b9827746c0b6.png)

![image](https://user-images.githubusercontent.com/31133476/148248155-484c38ce-5514-47db-b4e2-316f3008daf4.png)

**Comment by w13915984028** [comment]
NOTICE:
When Create VM, the "Node Scheduling" is "Run VM on any avail node" by default,  with first or third option, when there are more than 1 available qualified scheduling node (with enough resources), then when one node ,which runs the VM, goes down ,  the VM will be scheduled to another qualified scheduling node.

![image](https://user-images.githubusercontent.com/31133476/149024903-de7db646-4bbf-4625-ada3-623df937e26d.png)

**Comment by w13915984028** [comment]
In Harvester, both `virtualmachine `and `virtualmachineinstance `have a default value `LiveMigrate ` for `evictionStrategy`, 
@yasker  @guangbochen 

As per `KubeVirt `document and source code, when `LiveMigrate `is set, then when one NODE is shutdown, the VM on this NODE will be migrated to other nodes.

It explains why in Harvester v1.0.0, the VM will be migrated to other NODEs by default.


```
harv2:~ # kubectl get virtualmachines.kubevirt.io -A -o JSON
...
                        "evictionStrategy": "LiveMigrate",
                        "hostname": "cirros-3",
...

```

```
// +k8s:openapi-gen=true
type VirtualMachineInstanceSpec struct {
        // EvictionStrategy can be set to "LiveMigrate" if the VirtualMachineInstance should be
        // migrated instead of shut-off in case of a node drain.
        //
        // +optional
        EvictionStrategy *EvictionStrategy `json:"evictionStrategy,omitempty"`
```

**Comment by w13915984028** [comment]
The issue is solved/not existing in Harvester v1.0.0 and master-head now. 
VM will be rescheduled to other nodes when its current host node goes down, the pre-condition is that the VM´s "Node Scheduling" is "Run VM on any available node" and there are others nodes which can host this VM.

**Comment by noahgildersleeve** [comment]
Tested in daily master build. The VM will migrate based on `vm-force-reset-policy` 

## Environment setup
Setup an airgapped harvester
1. Create a 3 node harvester cluster 

## Verification Steps
1. Navigate to advanced settings and edit vm-force-reset-policy
![image](https://user-images.githubusercontent.com/83787952/146448317-a259d86d-2020-4bed-adc2-f19ecf0d3fbb.png)
1. Set reset policy to `60`
1. Create VM
1. Run health checks
1. Shut down node that is running VM
1. Check for when it starts to migrate to new Host

## Expected Results
1. It should migrate after 60 seconds
